lib/phoenix_swoosh.ex

```diff
@@ -236,4 +250,28 @@ defmodule Phoenix.Swoosh do
 
   defp template_name(name, format) when is_atom(name), do: Atom.to_string(name) <> "." <> format
   defp template_name(name, _format) when is_binary(name), do: name
+
+  @doc false
+  def __view__(swoosh_module) do
+    swoosh_module
+    |> Phoenix.Naming.unsuffix("Email")
```

Comment:
Maybe just leave it there. If we change `UserEmail` to `UserView` there is a good chance we have conflicts.

Reply:
Yes, you are right, it is possible, the idea is to provide simple and convenient default as of the Phoenix philosophy.

For example, I could have two modules: AccountsRecoveryController/AccountsRecoveryEmail and AccountsConfirmationController/AccountsConfirmationEmail; each could have a `show.html` view for the web request and an `email.html` view for the email.

But I could also organise my code into a single module called AccountsController/AccountsEmail grouping all the account related functionality, we'll need to organise our views in some way like having "confirmation.html" and "recovery.html". Or I could also have specific modules for email, so the email and web route views won't be shared the same view module.

Everything will be routed around Phoenix View template name argument, while still be open to other form of organization using annotations/use options such as Phoenix.Controller.

What do you think about this idea? I could also be completely wrong (^///^)

Reply:
A couple things:

1. https://github.com/phoenixframework/phoenix/pull/4373/files and https://github.com/phoenixframework/phoenix/pull/4368/files show that the phoenix convention for email modules will be called `Notifier`s
2. Notifier modules are more like context modules and less like controller modules. It actually doesn't make a lot of sense to mimic the controller behaviors.

Reply:
Mhhh, the idea of placing the notifier (mail) in the business context of the application makes sense in a way.

Following this idea, I'm not sure how to concretely organize the notifier with the Phoenix templates without polluting the application core business with the Web interface layer (hexagonal architecture).

Maybe we should use the `Phoenix.View` directly without Phoenix itself and use the template like `LiveView` does with a side file? In the current situation, I think that this PR does not make sense at the moment.

Reply:
Yes, the idea is to replace `phoenix` dependency with `phoenix_view`. So it both works in and outside of phoenix projects.